src/html.rs

```diff
@@ -194,7 +194,7 @@ impl<'s, I: Iterator<Item = Event<'s>>, W: std::fmt::Write> Writer<'s, I, W> {
 
                     for (a, v) in attrs.iter().filter(|(a, _)| *a != "class") {
                         write!(self.out, r#" {}=""#, a)?;
-                        self.write_escape(v)?;
+                        self.write_escape(&format!("{}", v))?;
```

Comment:
As we already encourage buffering on sinks sensitive to many small writes, I guess we can just use the `Attr::value` method and write each char (respecting escapes), like you propose?

Reply:
Fixed here:

https://github.com/hellux/jotdown/blob/7529059f47d1ab174cd1f4493f52dca68f9c61a5/src/html.rs#L197

Reply:
did a quick benchmark by trying to use the write_escaped_char in write_escape:
```diff
-    fn write_escape(&mut self, mut s: &str) -> std::fmt::Result {
-        let mut ent = "";
-        while let Some(i) = s.find(|c| {
-            match c {
-                '<' => Some("&lt;"),
-                '>' => Some("&gt;"),
-                '&' => Some("&amp;"),
-                _ => None,
-            }
-            .map_or(false, |s| {
-                ent = s;
-                true
-            })
-        }) {
-            self.out.write_str(&s[..i])?;
-            self.out.write_str(ent)?;
-            s = &s[i + 1..];
-        }
-        self.out.write_str(s)
+    fn write_escape(&mut self, s: &str) -> std::fmt::Result {
+        s.chars().try_for_each(|c| self.write_escaped_char(c))
     }
```

It seems like it is considerable slower to go char by char than trying to write strs (even when writing to buffered output):
```
  html/readme             time:   [39.416 µs 39.464 µs 39.540 µs]
                        thrpt:  [22.838 Melem/s 22.882 Melem/s 22.910 Melem/s]
                 change:
                        time:   [+76.224% +93.766% +127.16%] (p = 0.00 < 0.05)
                        thrpt:  [-55.979% -48.391% -43.254%]
```

However, it might be difficult to avoid to go char by char for the escapes (unless we expose the raw str?). Currently, the HTML output is around 10% slower on this branch compared to master, but the output is by far the fastest part so it is does not have a considerable impact, I think.

Reply:
We could make an iterator that iterates over string slices of `Attr::raw` instead? That seems similar to what you have done in the `inline_perf` branch. But as long as we only use `write_escaped_char` for writing the attributes, then we shouldn't be in such a bad shape, I agree.

Reply:
> We could make an iterator that iterates over string slices of `Attr::raw` instead? That seems similar to what you have done in the `inline_perf` branch.

yeah, could be worth a try

Reply:
Done!